include/acl.h

```diff
@@ -593,20 +593,19 @@ typedef struct acl_device_def_autodiscovery_t {
 } acl_device_def_autodiscovery_t;
 
 typedef struct acl_device_def_t {
-  // Back pointer. This is only valid if this def is owned by a dev_prog. In
-  // tests, this is not the case.
-  acl_device_program_info dev_prog;
```

Comment:
Do you know why this point exists in the first place? So from now on, dev_prog only lives in the `acl_device_binary_t` data structure now, correct?

Reply:
> Do you know why this point exists in the first place?

I might know at a point of time but I don't remember anymore :p

> So from now on, dev_prog only lives in the acl_device_binary_t data structure now, correct?

Based on my notes looks like the true copies of `acl_device_program_info` belongs to a `cl_program`, and the dev_prog in `acl_device_binary_t` is still just a back-pointer, but unlike the used back-pointer being removed in this change, it is used in acl_device_binary.cpp to get device, etc.